homeassistant/components/tesla_fleet/__init__.py

```diff
@@ -66,7 +67,10 @@ async def async_setup_entry(hass: HomeAssistant, entry: TeslaFleetConfigEntry) -
 
     async def _refresh_token() -> str:
         async with refresh_lock:
-            await oauth_session.async_ensure_token_valid()
+            try:
+                await oauth_session.async_ensure_token_valid()
+            except ClientResponseError as e:
+                raise ConfigEntryAuthFailed from e
```

Comment:
Shouldn't we check the status code (401 / 403) to know that it's an auth failure?

Reply:
The way I figure it, **any** sort of failure during a refresh token flow means the user is in an unauthenticated state. So even if the error isnt 401 / 403 the user is still going to need to run through the re-auth flow to get things working again.

Reply:
If the Tesla API does things correctly, other errors should not require a new authorization. Something else is broken then.

Reply:
Ok, raising ConfigEntryAuthFailed on 401 and ConfigEntryNotReady otherwise.